fuel/transformers/__init__.py

```diff
@@ -225,14 +225,17 @@ def get_data(self, request=None):
         if request is not None:
             raise ValueError
         data = next(self.child_epoch_iterator)
+        mapping_input = data
         if self.mapping_accepts == dict:
-            data = OrderedDict(equizip(self.data_stream.sources, data))
-        image = self.mapping(data)
+            mapping_input = OrderedDict(equizip(self.data_stream.sources,
+                                                data))
+        image = self.mapping(mapping_input)
+        image_sources = self.add_sources if self.add_sources else self.sources
```

Comment:
You probably mean `if not x is None`. My common sense tells me that the `else` body is exceptional and rare.

But it is not a big concern.